src/sortBy.ts

```diff
@@ -1,29 +1,59 @@
 import { purry } from './purry';
 
+type Direction = 'asc' | 'desc';
+type SortProjection<T> = (x: T) => Sortable;
+type SortablePrimitive = number | string;
+type Sortable = SortablePrimitive | { valueOf(): SortablePrimitive };
```

Comment:
two questions:

- what's the  object with the `valueOf` doing?
- previously, the function was allowed to return _any_, now we only allow to sort by numbers or strings I guess. I like that we are being more strict, but what about `booleans` for example? I think they have an order as well ...

Reply:
valueOf just says that any type that has a valueOf method is a valid type to use as a comparer.

I didn't realize boolean operands be used with non-equal comparison operators. I've added boolean to the list. Should be decently flexible though.